src/prometheus.ts

```diff
@@ -206,4 +68,131 @@ export async function loadPrometheusProvider(prometheusUrl: string) {
     setUrl,
     getUrl,
   };
+
+  /**
+   * Fetches the timeseries data for a given query and time range.
+   *
+   * NOTE - This is basically a copy of the `querySeries` function from fiberplane-prometheus-query.
+   *        At first, I needed to copy it here to fix an issue with the url.
+   *        But then I realized that we would need to polyfill fetch to use it,
+   *        so I think we'll just keep this version here for now, and
+   *        rely on the utilities to transform the request/response into the format we need.
```

Comment:
I think it might be a nice solution to add this to `fiberplane-prometheus-query` instead:

```ts
export type QueryPrometheusOptions = RequestInit & {
  /**
   * Base URL to the Prometheus service, without trailing slash.
   */
  baseUrl: string;
  
  /**
   * Custom `fetch` function to use instead of the global `fetch`.
   *
   * May be used to support environments that don't support `fetch`, or to inject `fetch` during tests.
   */
  fetch?: typeof fetch;
};
```

Reply:
yeah agreed — i was a little hesitant in this case because node-fetch is subtly different than browser fetch (e.g., it doesn't accept a cors parameter, but i don't think typescript would complain about that)